File: src/main/scala/info/crlog/higgs/protocols/boson/json/DemoServer.scala
package info.crlog.higgs.protocols.boson.json

import info.crlog.higgs.method
import json.BosonServer

//class Listener {
//  @method("test")
//  def test(a: Int)= {
//    println(a)
//    Array("blast")
//    new Message("test", Some(Seq("d")), Some("callback"), 103454)
//  }
//}

/**
 * @author Courtney Robinson <courtney@crlog.info>
 */
object DemoServer {
  def main(args: Array[String]) {
    val server = new BosonServer("localhost", 12001)
    server.register(new Listener)
    server.bind(() => {})
  }
}

File: higgs-boson/src/main/java/com/fillta/higgs/boson/demo/DemoServer.java
package com.fillta.higgs.boson.demo;

import com.fillta.higgs.boson.BosonServer;
import com.fillta.higgs.events.listeners.ChannelEventListener;
import com.google.common.base.Optional;
import io.netty.channel.ChannelHandlerContext;

/**
 * @author Courtney Robinson <courtney@crlog.info>
 */
public class DemoServer {
    protected DemoServer() {
    }

    public static void main(String... args) {
        BosonServer server = new BosonServer(8080);
        server.onException(new ChannelEventListener() {
            public void triggered(ChannelHandlerContext ctx, Optional<Throwable> ex) {
                ex.get().printStackTrace();
            }
        });
        server.registerPackage(Listener.class.getPackage().getName());
        server.bind();
    }
}
